.gitignore

```diff
@@ -0,0 +1,17 @@
+*.class
+.gradle
+/build/
+/bin/
+/.settings/
+.project
+.classpath
+
+# Package Files
+*.jar
+*.war
+*.ear
+
+# Serverless directories
+.serverless
+
+!/gradle/wrapper/gradle-wrapper.jar
```

Comment:
for whatever reason i hadn't seen `!` used in a .gitignore before, but it makes sense since `*.jar` is ignored above. something new!